src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs

```diff
@@ -106,14 +106,18 @@ public static ImmutableArray<T> CreateRange<T>(IEnumerable<T> items)
             var immutableArray = items as IImmutableArray;
             if (immutableArray != null)
             {
-                immutableArray.ThrowInvalidOperationIfNotInitialized();
+                Array array = immutableArray.Array;
+                if (array == null)
+                {
+                    throw new InvalidOperationException(SR.InvalidOperationOnDefaultArray);
+                }
```

Comment:
No need to get numbers for me. Avoiding interface dispatch sounds worthy. Thanks.